src/cmap/options.rs

```diff
@@ -98,3 +98,37 @@ impl ConnectionPoolOptions {
             .build()
     }
 }
+
+/// Options used for constructing a `Connection`.
```

Comment:
I moved `ConnectionOptions` here so that all of the CMAP options structs (`StreamOptions`, `ConnectionOptions`, and `ConnectionPoolOptions` are in the same place). I feel like this will make it easier to remember what import path to use when trying to import one of them, but let me know if you disagree and we can come out with another solution.